nginx/1.25.3-zlib-ng.patch

```diff
@@ -0,0 +1,332 @@
++        ngx_feature_libs="-lz-ng"
++        ngx_feature_test="zng_stream z; zng_deflate(&z, Z_NO_FLUSH)"
+         . auto/feature
+ 
+ 
+diff -urN nginx-1.25.3.orig/src/http/modules/ngx_http_gunzip_filter_module.c nginx-1.25.3/src/http/modules/ngx_http_gunzip_filter_module.c
+--- nginx-1.25.3.orig/src/http/modules/ngx_http_gunzip_filter_module.c	2023-10-24 15:46:47.000000000 +0200
++++ nginx-1.25.3/src/http/modules/ngx_http_gunzip_filter_module.c	2023-11-25 04:00:16.017729654 +0100
+@@ -10,7 +10,14 @@
+ #include <ngx_core.h>
+ #include <ngx_http.h>
+ 
+-#include <zlib.h>
++#if defined(NGX_ZLIB_NG)
++# include <zlib-ng.h>
++# define ZPREFIX(x) zng_ ## x
++# define z_stream zng_stream
++#elif defined(NGX_ZLIB)
++# include <zlib.h>
++# define ZPREFIX(x) x
++#endif
+ 
+ 
+ typedef struct {
+@@ -312,7 +319,7 @@
+     ctx->zstream.opaque = ctx;
+ 
+     /* windowBits +16 to decode gzip, zlib 1.2.0.4+ */
+-    rc = inflateInit2(&ctx->zstream, MAX_WBITS + 16);
++    rc = ZPREFIX(inflateInit2)(&ctx->zstream, MAX_WBITS + 16);
+ 
+     if (rc != Z_OK) {
+         ngx_log_error(NGX_LOG_ALERT, r->connection->log, 0,
+@@ -425,7 +432,7 @@
+                    ctx->zstream.avail_in, ctx->zstream.avail_out,
+                    ctx->flush, ctx->redo);
+ 
+-    rc = inflate(&ctx->zstream, ctx->flush);
++    rc = ZPREFIX(inflate)(&ctx->zstream, ctx->flush);
+ 
+     if (rc != Z_OK && rc != Z_STREAM_END && rc != Z_BUF_ERROR) {
+         ngx_log_error(NGX_LOG_ERR, r->connection->log, 0,
+@@ -523,7 +530,7 @@
+ 
+     if (rc == Z_STREAM_END && ctx->zstream.avail_in > 0) {
+ 
+-        rc = inflateReset(&ctx->zstream);
++        rc = ZPREFIX(inflateReset)(&ctx->zstream);
+ 
+         if (rc != Z_OK) {
+             ngx_log_error(NGX_LOG_ALERT, r->connection->log, 0,
+@@ -574,7 +581,7 @@
+     ngx_log_debug0(NGX_LOG_DEBUG_HTTP, r->connection->log, 0,
+                    "gunzip inflate end");
+ 
+-    rc = inflateEnd(&ctx->zstream);
++    rc = ZPREFIX(inflateEnd)(&ctx->zstream);
+ 
+     if (rc != Z_OK) {
+         ngx_log_error(NGX_LOG_ALERT, r->connection->log, 0,
+diff -urN nginx-1.25.3.orig/src/http/modules/ngx_http_gzip_filter_module.c nginx-1.25.3/src/http/modules/ngx_http_gzip_filter_module.c
+--- nginx-1.25.3.orig/src/http/modules/ngx_http_gzip_filter_module.c	2023-10-24 15:46:47.000000000 +0200
++++ nginx-1.25.3/src/http/modules/ngx_http_gzip_filter_module.c	2023-11-25 04:01:03.754887799 +0100
+@@ -9,7 +9,14 @@
+ #include <ngx_core.h>
+ #include <ngx_http.h>
+ 
+-#include <zlib.h>
++#if defined(NGX_ZLIB_NG)
++# include <zlib-ng.h>
++# define ZPREFIX(x) zng_ ## x
++# define z_stream zng_stream
++#elif defined(NGX_ZLIB)
++# include <zlib.h>
++# define ZPREFIX(x) x
++#endif
+ 
+ 
+ typedef struct {
+@@ -56,8 +63,7 @@
+     unsigned             done:1;
+     unsigned             nomem:1;
+     unsigned             buffering:1;
+-    unsigned             zlib_ng:1;
+-    unsigned             state_allocated:1;
++    unsigned             intel:1;
```

Comment:
Why replace `zlib_n` with `intel` here and several following places? This makes the diff bigger for no reason.